File: output/CodeGen/avatar/Java/Python/codeforces_334_A.py
class codeforces_334_A {
    def main(args):
        in = input()
        print(in.nextInt() * in.nextInt())
        square = in.nextInt() * in.nextInt()
        odd = []
        even = []
        for i in range(1, square+1):
            if i % 2 == 0:
                even.append(i)
            else:
                odd.append(i)
        loop = square / in.nextInt()
        div = loop / 2
        for i in range(1, loop+1):
            for j in range(0, div):
                if i % 2 == 1:
                    print(odd.pop(0) + " " + even.pop(even.size() - 1))
                else:
                    print(even.pop(0) + " " + odd.pop(odd.size() - 1))
            print()
        print()


}



Input:

3
3
1 2 3
2 4 6
3 5 7

Output:

12
36

File: output/ChatGPT/avatar/Java/C++/codeforces_334_A.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;
    int square = n * n;
    vector<int> odd;
    vector<int> even;
    
    for (int i = 1; i <= square; i++) {
        if (i % 2 == 0) {
            even.push_back(i);
        } else {
            odd.push_back(i);
        }
    }
    
    int loop = square / n;
    int div = loop / 2;
    
    for (int i = 1; i <= loop; i++) {
        for (int j = 0; j < div; j++) {
            if (i % 2 == 1) {
                cout << odd[0] << " " << even[even.size() - 1] << " ";
                odd.erase(odd.begin());
                even.pop_back();
            } else {
                cout << even[0] << " " << odd[odd.size() - 1] << " ";
                even.erase(even.begin());
                odd.pop_back();
            }
        }
        cout << endl;
    }
    
    return 0;
}